src/app/api/v1/user/welcome-scan/create/route.ts

```diff
@@ -0,0 +1,89 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { getServerSession } from "next-auth";
+import { authOptions } from "../../../../utils/auth";
+import { NextRequest, NextResponse } from "next/server";
+
+import {
+  createProfile,
+  createScan,
+  isEligible,
+} from "../../../../../functions/server/onerep";
+import type { ProfileData } from "../../../../../functions/server/onerep";
+import AppConstants from "../../../../../../appConstants";
+import { getSubscriberByEmail } from "../../../../../../db/tables/subscribers";
+
+import {
+  setOnerepProfileId,
+  setOnerepScan,
+} from "../../../../../../db/tables/onerep_scans";
+import { setProfileDetails } from "../../../../../../db/tables/onerep_profiles";
+import { StateAbbr } from "../../../../../../utils/states";
+
+export interface WelcomeScanBody {
+  success: boolean;
+}
+
+export async function POST(
+  req: NextRequest
+): Promise<NextResponse<WelcomeScanBody | unknown>> {
+  const session = await getServerSession(authOptions);
+
+  const eligible = await isEligible();
+  if (!eligible) {
+    throw new Error("User has no more manual scans left");
+  }
+
+  const params = await req.json();
```

Comment:
We can create an interface here for the required params?

Reply:
They are checked right below, maybe using an interface and type guard for the runtime check would be a better pattern

Reply:
Yes! Moved the interface over from `EnterInfo` and sharing it now.